src/rules/semicolonRule.ts

```diff
@@ -171,11 +172,20 @@ class SemicolonWalker extends Lint.RuleWalker {
         const always = !never && (this.hasOption(OPTION_ALWAYS) || (this.getOptions() && this.getOptions().length === 0));
 
         if (always && !hasSemicolon) {
-            const failureStart = Math.min(position, this.getLimit());
-            const fix = new Lint.Fix(Rule.metadata.ruleName, [
-                this.appendText(failureStart, ";"),
-            ]);
-            this.addFailureAt(failureStart, 0, Rule.FAILURE_STRING_MISSING, fix);
+            const lastChild = children[children.length - 1];
```

Comment:
I don't think that there are valid TS code where `PropertySignature` doesn't have any child nodes (it should have name at least). Even if it might be `undefined` in other cases, it will be accessed only inside `InterfaceDeclaration` where it can't be valid (I haven't find any such cases).